presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJdbcIntegrationSmokeTest.java

```diff
@@ -25,4 +26,62 @@ public TestJdbcIntegrationSmokeTest()
     {
         super(() -> createJdbcQueryRunner(ORDERS));
     }
+
+    @Test
+    public void testPlanOptimizerFilterPushdownCorrectness()
+    {
+        assertUpdate("CREATE TABLE tpch.test_pushdown(col1 bigint, col2 bigint)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(5, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(2, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(4, 4)", "VALUES(1)");
+        // PickTableLayout#pushPredicateIntoTableScan cannot push down this predicate however, JdbcPlanOptimizer can.
+        assertQuery("SELECT * FROM tpch.test_pushdown WHERE col1 + col2 = 8", "VALUES(4, 4)");
+        assertUpdate("DROP TABLE tpch.test_pushdown");
+    }
+
+    @Test
+    public void testPlanOptimizerAndDefaultFilterPushdownCorrectness()
+    {
+        assertUpdate("CREATE TABLE tpch.test_pushdown(col1 bigint, col2 bigint)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(5, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(2, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(4, 4)", "VALUES(1)");
+        // PickTableLayout#pushPredicateIntoTableScan and JdbcPlanOptimizer are both capable of pushing down these predicates.
+        assertQuery("SELECT * FROM tpch.test_pushdown WHERE col1 = 5 AND col2 = 4", "VALUES(5, 4)");
+        assertUpdate("DROP TABLE tpch.test_pushdown");
+    }
+
+    @Test
+    public void testIncompatibleFilterPushdownCorrectness()
+    {
+        assertUpdate("CREATE TABLE tpch.test_pushdown(col1 bigint, col2 bigint)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(5, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(2, 4)", "VALUES(1)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(4, 4)", "VALUES(1)");
+        // Neither PickTableLayout#pushPredicateIntoTableScan or JdbcPlanOptimizer are capable of pushing down these predicates.
+        assertQuery("SELECT * FROM tpch.test_pushdown WHERE NOT(col1 + col2 <= 6)", "VALUES (5, 4), (4, 4)");
+        assertUpdate("DROP TABLE tpch.test_pushdown");
+    }
+
+    @Test
+    public void testArithmetic()
+    {
+        assertUpdate("CREATE TABLE tpch.test_pushdown(col1 bigint, col2 bigint)");
+        assertUpdate("INSERT INTO tpch.test_pushdown VALUES(5, 4)", "VALUES(1)");
```

Comment:
You can insert multiple lines in one batch. See `testAlterTable` in `TestRaptorIntegrationSmokeTest` for example.